src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -278,11 +278,16 @@ private static string GetIdentifierFromPath(string filePath)
 
             for (var i = 0; i < filePath.Length; i++)
             {
-                builder.Append(filePath[i] switch
+                switch (filePath[i])
                 {
-                    ':' or '\\' or '/' => '_',
-                    var @default => @default,
-                });
+                    case ':' or '\\' or '/':
+                    case char ch when !char.IsLetterOrDigit(ch):
+                        builder.Append('_');
+                        break;
+                    default:
+                        builder.Append(filePath[i]);
+                        break;
+                }
             }
 
             return builder.ToString();
```

Comment:
So this is essentially used as a cache key?

Reply:
Yeah, this gets used as a the hint name for the generated SourceText. It's the filename that gets written to disk when the `/p:EmitCompilerGeneratedFiles=true` is set true and is used as the identifier for the generated file.